tests/integrations/test_notebooks.py

```diff
@@ -44,6 +49,7 @@ def _notebook_run(path, timeout=default_timeout):
 
         fout.seek(0)
         nb = nbformat.read(fout, nbformat.current_nbformat)
+    os.remove(fname)
```

Comment:
this will throw an OSError on non NT systems, right? Because the tempfile does not exist anymore?

Reply:
I added `delete=False` to the tempfile constructor (see with-statement), so the file will exist on all systems before that `remove()` call. Unfortunately, I cannot test it in other systems.

Reply:
ok. Why do you remove the file outside of the context manager?

couldn't you do something like 

``` Python
    with tempfile.NamedTemporaryFile(mode='w+t', suffix=".ipynb", delete=False) as fout:
        args = ["jupyter", "nbconvert",
                "--to", "notebook", "--execute", execproc_timeout]
        if six.PY2:
            args += ["--ExecutePreprocessor.kernel_name=python2"]
        elif six.PY3:
            args += ["--ExecutePreprocessor.kernel_name=python3"]
        args += ["--output", fout.name, path]
        subprocess.check_call(args)

        fout.seek(0)
        nb = nbformat.read(fout, nbformat.current_nbformat)
        os.remove(fout.name)
```

Reply:
I believe I'm not explaining it that well. Basically, NT does not allow you to delete a file which is currently being used by a process (used also means opened for read-only, etc). Not only that, but NT does not allow the same process to delete the file without first explicitly closing it. Therefore, the `close()` method must be first invoked - this happens automatically at the end of the with-statement. Now, the `remove()` method can only be called after the `close()` method, so it must be outside the context manager. Hope this explains it.

Reply:
Wow. Thank you for the detailed explanation, I haven't used windows in like 10-15 years, so these problems are completely new to me. 😃 

It blows my mind that the `NamedTemporaryFile` method is broken under windows NT. I googled around and your solution seem to be the prefered way to fix this.

Just out of curiosity, does this improve in python3?